Classes/IndexQueue/Indexer.php

```diff
@@ -377,7 +377,10 @@ protected function itemToDocument(Item $item, int $language = 0): ?Document
         $document = null;
 
         $itemRecord = $this->getFullItemRecord($item, $language);
-        if (!is_null($itemRecord)) {
+
+        $isVisible = ((is_array($itemRecord) && empty($itemRecord['hidden'])));
```

Comment:
Please see the patch file of https://github.com/TYPO3-Solr/ext-solr/issues/4409#issuecomment-3307127824
The TCA defintion for "hidden" must be used instead of hard-coded `['hidden']` array access key.